@here/harp-mapview/lib/Utils.ts

```diff
@@ -242,53 +243,194 @@ export namespace MapViewUtils {
+        // at the mapTargetWorld, then choose a rotation center target above the mapTargetWorld,
+        // i.e. offsetY > 0. In such case, we just return 0, i.e. we don't let the user increase
+        // the tilt (but it can decrease, see check above for "deltaTilt <= 0").
+        if (rotationCenterTilt > maxRotationTiltAngle) {
+            return 0;
         }
 
-        const posBackup = cache.vector3[1].copy(camera.position);
-        const quatBackup = cache.quaternions[1].copy(camera.quaternion);
+        // This is used to find the max tilt angle, because the difference in normals is needed
+        // to correct the triangle used to find the max tilt angle at the rotation center.
+        let angleBetweenNormals = 0;
+        if (projection === sphereProjection) {
+            const projectedRotationTargetNormal = projection
+                .surfaceNormal(rotationTargetWorld, cache.vector3[0])
+                .projectOnPlane(tiltAxis);
+            const mapTargetNormal = projection.surfaceNormal(mapTargetWorld, cache.vector3[1]);
+            angleBetweenNormals = projectedRotationTargetNormal.angleTo(mapTargetNormal);
+        }
+
+        const ninetyRad = THREE.MathUtils.degToRad(90);
+        // The following terminology will be used:
+        // Ta = Tilt axis, tilting is achieved by rotating the camera around this point.
+        // R = rotation target, i.e. the point about which we are rotating: `rotationTargetWorld`
+        // Rp = rotation target projected on to the plane defined with normal as the tilt axis.
+        // C = camera position
+        // M = map target, i.e. the point which the camera is looking at: `mapTargetWorld`
+        // Note, the points Rp, C, and M create a plane that is perpendicular to the earths surface,
+        // because the tilt axis is perpendicular to the up vector.
+        // The following variable `angleBetweenRotationTargetAndMapTarget` is the angle between the
+        // two rays C->Rp and C->M, i.e. the angle RpCM. This angle remains constant when tilting with
+        // a fixed `offsetX` and `offsetY`. It is calculated by using the intersection of the two
+        // ways with the earth. Where:
+        // MRpC is 90 + `angleBetweenNormals` - tilt at Rp.
+        // CMRp is 90 + `tilt`
+        // RpCM is 180 - CRpM - RpMC, aka `angleBetweenRotationTargetAndMapTarget`
+        // Note the use of `angleBetweenNormals` to ensure this works for spherical projections.
+        // Note, also, the above calculation only works when the tilt at the M is less than the tilt
+        // at Rp, otherwise the above formula won't work. We however don't need to worry about this
+        // case because this happens only when offsetY is less than zero, and this is handled above.
+        const angleBetweenRotationTargetAndMapTarget =
+            ninetyRad * 2 -
+            (ninetyRad + angleBetweenNormals - rotationCenterTilt + ninetyRad + tilt);
```

Comment:
Instead of explaining in comments above how to compute the angles, just write constants with the angle names. I rather see self-explanatory code than lengthy comments. For example:

```suggestion
        const MRC = 90 + angleBetweenNormals - tilt;
        const CMR = 90 + tilt;
        const RCM = 180 - MRC - CMR;
```

Reply:
Agreed, will do.